diff --git a/LogicCircuitSimulator/CreateConnectionDialog.Designer.cs b/LogicCircuitSimulator/CreateConnectionDialog.Designer.cs
index 0b0661b..5107955 100644
--- a/LogicCircuitSimulator/CreateConnectionDialog.Designer.cs
+++ b/LogicCircuitSimulator/CreateConnectionDialog.Designer.cs
@@ -29,13 +29,33 @@
         private void InitializeComponent()
         {
             System.ComponentModel.ComponentResourceManager resources = new System.ComponentModel.ComponentResourceManager(typeof(CreateConnectionDialog));
+            this.listBox1 = new System.Windows.Forms.ListBox();
+            this.listBox2 = new System.Windows.Forms.ListBox();
             this.SuspendLayout();
             // 
+            // listBox1
+            // 
+            this.listBox1.FormattingEnabled = true;
+            this.listBox1.Location = new System.Drawing.Point(12, 76);
+            this.listBox1.Name = "listBox1";
+            this.listBox1.Size = new System.Drawing.Size(108, 95);
+            this.listBox1.TabIndex = 0;
+            // 
+            // listBox2
+            // 
+            this.listBox2.FormattingEnabled = true;
+            this.listBox2.Location = new System.Drawing.Point(163, 76);
+            this.listBox2.Name = "listBox2";
+            this.listBox2.Size = new System.Drawing.Size(120, 95);
+            this.listBox2.TabIndex = 1;
+            // 
             // CreateConnectionDialog
             // 
             this.AutoScaleDimensions = new System.Drawing.SizeF(6F, 13F);
             this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
             this.ClientSize = new System.Drawing.Size(295, 294);
+            this.Controls.Add(this.listBox2);
+            this.Controls.Add(this.listBox1);
             this.Icon = ((System.Drawing.Icon)(resources.GetObject("$this.Icon")));
             this.Name = "CreateConnectionDialog";
             this.Text = "CreateConnection";
@@ -44,5 +64,8 @@
         }
 
         #endregion
+
+        private System.Windows.Forms.ListBox listBox1;
+        private System.Windows.Forms.ListBox listBox2;
     }
 }
\ No newline at end of file
diff --git a/LogicCircuitSimulator/MainForm.cs b/LogicCircuitSimulator/MainForm.cs
index 4e5e556..a1195c1 100644
--- a/LogicCircuitSimulator/MainForm.cs
+++ b/LogicCircuitSimulator/MainForm.cs
@@ -140,57 +140,7 @@ namespace LogicCircuitSimulator
 
 
 
-        private void PicPox_Move(object sender, EventArgs e)
-        {
-            //int _senderTag = (int)(((PictureBox)sender).Tag);
-                panel1.Invalidate();
-        }
-
-        private void PicBox_MouseMove(object sender, MouseEventArgs e)
-        {
-            if (e.Button == MouseButtons.Left
-            && MouseDownOnSelectedPicBox
-            )
-            {
-                ((PictureBox)sender).Cursor = Cursors.SizeAll;
-                ((PictureBox)sender).Left = e.X + ((PictureBox)sender).Left - MouseDownLocation.X;
-                ((PictureBox)sender).Top = e.Y + ((PictureBox)sender).Top - MouseDownLocation.Y;
-            }
-        }
-
-        private void PicBox_MouseUp(object sender, MouseEventArgs e)
-        {
-            if ((sender as PictureBox).Cursor != Cursors.Default)
-                (sender as PictureBox).Cursor = Cursors.Default;
-
-            MouseDownOnSelectedPicBox = false;
-        }
-
-        private void PicBox_MouseDown(object sender, MouseEventArgs e)
-        {
-            if (e.Button == MouseButtons.Left)
-            {
-                MouseDownOnSelectedPicBox = true;
 
-                MouseDownLocation = e.Location;
-            }
-        }
-
-        private void PicBox_MouseLeave(object sender, EventArgs e)
-        {
-            if (ModifierKeys == Keys.Control && Connecting)
-            {
-                ReceiveingControlTag = -1;
-            }
-        }
-
-        private void PicBox_MouseEnter(object sender, EventArgs e)
-        {
-            if (ModifierKeys == Keys.Control && Connecting)
-            {
-                ReceiveingControlTag = (int)(((Control)sender).Tag);
-            }
-        }
 
         //private void PicBox_Paint(object sender, PaintEventArgs e)
         //{
@@ -257,6 +207,58 @@ namespace LogicCircuitSimulator
             panel1.Controls.Add(pb);*/
         }
 
+        private void PicPox_Move(object sender, EventArgs e)
+        {
+            //int _senderTag = (int)(((PictureBox)sender).Tag);
+            panel1.Invalidate();
+        }
+
+        private void PicBox_MouseMove(object sender, MouseEventArgs e)
+        {
+            if (e.Button == MouseButtons.Left
+            && MouseDownOnSelectedPicBox
+            )
+            {
+                ((PictureBox)sender).Cursor = Cursors.SizeAll;
+                ((PictureBox)sender).Left = e.X + ((PictureBox)sender).Left - MouseDownLocation.X;
+                ((PictureBox)sender).Top = e.Y + ((PictureBox)sender).Top - MouseDownLocation.Y;
+            }
+        }
+
+        private void PicBox_MouseUp(object sender, MouseEventArgs e)
+        {
+            if ((sender as PictureBox).Cursor != Cursors.Default)
+                (sender as PictureBox).Cursor = Cursors.Default;
+
+            MouseDownOnSelectedPicBox = false;
+        }
+
+        private void PicBox_MouseDown(object sender, MouseEventArgs e)
+        {
+            if (e.Button == MouseButtons.Left)
+            {
+                MouseDownOnSelectedPicBox = true;
+
+                MouseDownLocation = e.Location;
+            }
+        }
+
+        private void PicBox_MouseLeave(object sender, EventArgs e)
+        {
+            if (ModifierKeys == Keys.Control && Connecting)
+            {
+                ReceiveingControlTag = -1;
+            }
+        }
+
+        private void PicBox_MouseEnter(object sender, EventArgs e)
+        {
+            if (ModifierKeys == Keys.Control && Connecting)
+            {
+                ReceiveingControlTag = (int)(((Control)sender).Tag);
+            }
+        }
+
         private void bClear_Click(object sender, EventArgs e)
         {
             if (MessageBox.Show("Are you sure?\nThis will delete all the elements!"
